main.go

```diff
@@ -29,6 +32,44 @@ func main() {
 	w.Resize(fyne.NewSize(300, 500))
 	w.SetIcon(resourceIconPng)
 
+	if deskApp, ok := a.(desktop.App); ok {
+		w.SetCloseIntercept(func() {
+			w.Hide() // don't close the window if system tray used
+		})
```

Comment:
You don't need to wrap the function call. You can just do the following:
```suggestion
		w.SetCloseIntercept(w.Hide) // don't close the window if system tray used
```